engine/src/main/java/org/hibernate/search/cfg/spi/IndexManagerFactory.java

```diff
@@ -15,20 +15,27 @@
  * define new short-hand aliases, change the default implementation.
  *
  * @author Sanne Grinovero (C) 2012 Red Hat Inc.
+ * @author Guillaume Smet
  */
 public interface IndexManagerFactory extends Service {
 
 	/**
-	 * @return a new instance of the default IndexManager
+	 * Determine the {@code IndexManager} implementation which will be used for this entity type.
+	 *
+	 * @param mappedClass the mapped entity class
+	 * @param indexManagerImplementationName how this is resolved to an {@code IndexManager} type
+	 * is left to the implementor.
+	 *
+	 * @return the type chosen
 	 */
-	IndexManager createDefaultIndexManager();
+	Class<? extends IndexManager> determineIndexManagerImpl(Class<?> mappedClass, String indexManagerImplementationName);
```

Comment:
How comes the mapped class is needed here, too? It seems to be used for caching, but couldn't this happen using the implementation name as a key? Same for the method below, the implementation I could find does not use the `mappedClass` at all.